winpr/libwinpr/thread/thread.c

```diff
@@ -285,12 +285,13 @@ static void winpr_StartThread(WINPR_THREAD *thread)
 	if (thread->dwStackSize > 0)
 		pthread_attr_setstacksize(&attr, (size_t) thread->dwStackSize);
 
-	pthread_create(&thread->thread, &attr, thread_launcher, thread);
-	pthread_attr_destroy(&attr);
+	thread->started = TRUE;
 	reset_event(thread);
+
+	pthread_create(&thread->thread, &attr, thread_launcher, thread);
 	ListDictionary_Add(thread_list, &thread->thread, thread);
```

Comment:
The list is synchronized, so it automatically locks.

On 2015-01-16 14:40, Hardening wrote:
>
> In winpr/libwinpr/thread/thread.c
> <https://github.com/FreeRDP/FreeRDP/pull/2314#discussion-diff-23080294>:
>
> >  	reset_event(thread);
> > +
> > +	pthread_create(&thread->thread, &attr, thread_launcher, thread);
> >  	ListDictionary_Add(thread_list, &thread->thread, thread);
>
> Stupid question: thread_list is accessed and modified without taking
> any lock ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeRDP/FreeRDP/pull/2314/files#r23080294>.
>